src/dtos/duct-waf-base.dto.ts

```diff
@@ -1,48 +1,100 @@
+    },
+  })
   wafDeterminationDate: Date;
 
   @ApiProperty({
     description: propertyMetadata.ductWafDTOWafBeginDate.description,
     example: propertyMetadata.ductWafDTOWafBeginDate.example,
     name: propertyMetadata.ductWafDTOWafBeginDate.fieldLabels.value,
   })
+  @IsIsoFormat({
+    message: (args: ValidationArguments) => {
+      return `${args.property} [RECTDUCTWAF-FATAL-A] The value for ${args.value} in the Rectangular Duct Waf record ${args.property} must be a valid ISO date format yyyy-mm-dd`;
+    },
+  })
   wafBeginDate: Date;
 
   @ApiProperty({
     description: propertyMetadata.ductWafDTOWafBeginHour.description,
     example: propertyMetadata.ductWafDTOWafBeginHour.example,
     name: propertyMetadata.ductWafDTOWafBeginHour.fieldLabels.value,
   })
+  @IsInt()
+  @IsInRange(0, 23, {
+    message: (args: ValidationArguments) => {
+      return `${args.property} [RECTDUCTWAF-FATAL-A] The value for ${args.value} in the Rectangular Duct Waf record ${args.property} must be within the range of 0 and 23`;
+    },
+  })
   wafBeginHour: number;
 
   @ApiProperty({
     description: propertyMetadata.ductWafDTOWafMethodCode.description,
     example: propertyMetadata.ductWafDTOWafMethodCode.example,
     name: propertyMetadata.ductWafDTOWafMethodCode.fieldLabels.value,
   })
+  @IsInDbValues(
+    'SELECT waf_method_cd as "value" FROM camdecmpsmd.waf_method_code',
+    {
+      message: (args: ValidationArguments) => {
+        return `${args.property} [RECTDUCTWAF-FATAL-B] The value for ${args.value} in the Rectangular Duct Waf record ${args.property} is invalid`;
+      },
+    },
+  )
   wafMethodCode: string;
 
   @ApiProperty({
     description: propertyMetadata.ductWafDTOWafValue.description,
     example: propertyMetadata.ductWafDTOWafValue.example,
     name: propertyMetadata.ductWafDTOWafValue.fieldLabels.value,
   })
+  @IsNumber(
+    { maxDecimalPlaces: 4 },
+    {
+      message: (args: ValidationArguments) => {
+        return `${args.property} [RECTDUCTWAF-FATAL-A] The value for ${args.value} in the Rectangular Duct Waf record ${args.property} is allowed only 4 decimal place`;
+      },
+    },
+  )
+  @IsInRange(-99.9999, 99.9999, {
+    message: (args: ValidationArguments) => {
+      return `${args.property} [RECTDUCTWAF-FATAL-A] The value for ${args.value} in the Rectangular Duct Waf record ${args.property} must be within the range of -99.9999 and 99.9999`;
+    },
+  })
   wafValue: number;
 
   @ApiProperty({
     description: propertyMetadata.ductWafDTONumberOfTestRuns.description,
     example: propertyMetadata.ductWafDTONumberOfTestRuns.example,
     name: propertyMetadata.ductWafDTONumberOfTestRuns.fieldLabels.value,
   })
+  @IsInt()
+  @IsAtMostDigits(2, {
+    message: (args: ValidationArguments) => {
+      return `${args.property} [RECTDUCTWAF-FATAL-A] The value for ${args.value} in the Rectangular Duct Waf record ${args.property} must be 5 digits or less`;
```

Comment:
change error message to 2 digits or less

Reply:
✔